crates/bevy_math/src/bounding/bounded3d.rs

```diff
@@ -0,0 +1,286 @@
+}
+
+#[test]
+fn test_aabb3d_contains() {
+    let a = Aabb3d {
+        min: Vec3::new(-1., -1., -1.),
+        max: Vec3::new(1., 1., 1.),
+    };
+    let b = Aabb3d {
+        min: Vec3::new(-2., -1., -1.),
+        max: Vec3::new(1., 1., 1.),
+    };
+    assert!(!a.contains(&b));
+    let b = Aabb3d {
+        min: Vec3::new(-0.25, -0.8, -0.9),
+        max: Vec3::new(1., 1., 0.9),
+    };
+    assert!(a.contains(&b));
+}
+
+#[test]
+fn test_aabb3d_merged() {
+    let a = Aabb3d {
+        min: Vec3::new(-1., -1., -1.),
+        max: Vec3::new(1., 0.5, 1.),
+    };
+    let b = Aabb3d {
+        min: Vec3::new(-2., -0.5, -0.),
+        max: Vec3::new(0.75, 1., 2.),
+    };
+    let merged = a.merged(&b);
+    assert!((merged.min - Vec3::new(-2., -1., -1.)).length() < 0.0001);
+    assert!((merged.max - Vec3::new(1., 1., 2.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb3d_padded() {
+    let a = Aabb3d {
+        min: Vec3::new(-1., -1., -1.),
+        max: Vec3::new(1., 1., 1.),
+    };
+    let padded = a.padded((Vec3::ONE, Vec3::Y));
+    assert!((padded.min - Vec3::new(-2., -2., -2.)).length() < 0.0001);
+    assert!((padded.max - Vec3::new(1., 2., 1.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb3d_shrunk() {
+    let a = Aabb3d {
+        min: Vec3::new(-1., -1., -1.),
+        max: Vec3::new(1., 1., 1.),
+    };
+    let shrunk = a.shrunk((Vec3::ONE, Vec3::Y));
+    assert!((shrunk.min - Vec3::new(-0., -0., -0.)).length() < 0.0001);
+    assert!((shrunk.max - Vec3::new(1., 0., 1.)).length() < 0.0001);
+}
+
+/// A bounding sphere
+pub struct BoundingSphere {
+    /// The center of the bounding sphere
+    center: Vec3,
+    /// The radius of the bounding sphere
+    radius: f32,
+}
+
+impl BoundingVolume for BoundingSphere {
+    type Position = Vec3;
+    type Padding = f32;
+
+    #[inline(always)]
+    fn center(&self) -> Self::Position {
+        self.center
+    }
+
+    #[inline(always)]
+    fn visible_area(&self) -> f32 {
+        2. * std::f32::consts::PI * self.radius * self.radius
+    }
+
+    #[inline(always)]
+    fn contains(&self, other: &Self) -> bool {
+        if other.center == self.center {
+            other.radius <= self.radius
+        } else {
+            let furthest_point = (other.center - self.center).length() + other.radius;
+            furthest_point <= self.radius
+        }
+    }
+
+    #[inline(always)]
+    fn merged(&self, other: &Self) -> Self {
+        if other.center == self.center {
+            Self {
+                center: self.center,
+                radius: self.radius.max(other.radius),
+            }
+        } else {
+            let diff = other.center - self.center;
+            let length = diff.length();
+            let dir = diff / length;
+
+            Self {
+                center: self.center + dir * ((length + other.radius - self.radius) / 2.),
+                radius: (length + self.radius + other.radius) / 2.,
+            }
+        }
+    }
```

Comment:
Gave it a shot and tried fixing the logic bug, haven't tested.

```suggestion
    #[inline(always)]
    fn merged(&self, other: &Self) -> Self {
        let diff = other.center - self.center;
        let length = diff.length();
        if self.radius >= length + other.radius {
            return self;
        }
        if other.radius >= length + self.radius {
            return other;
        }
        let dir = diff / length;

        Self {
            center: (self.center + other.center) / 2. + dir * ((other.radius - self.radius) / 2.),
            radius: (length + self.radius + other.radius) / 2.,
        }
    }
```

Note that this also slightly helps the situation with spheres that are very close, because unless their radii are also nearly the same, one is going to contain the other, dodging that division entirely. In other words, getting past the first two ifs without returning means `length > |self.radius - other.radius|`.

Division by zero is still maybe possible though, I would consider adding a special case for when 
`(self.radius - other.radius).abs() < EPSILON` where the returned value is:
```rust
        Self {
            center: (self.center + other.center) / 2., // the term containing `other.radius - self.radius` and the direction is aproximated away
            radius: (length + self.radius + other.radius) / 2.,
        }
```